mlflow/deployments/base.py

```diff
@@ -70,6 +70,226 @@ def target_help():
+                              name, raises a :py:class:`mlflow.exceptions.MlflowException`.
+        :param config: (optional) Dict containing target-specific configuration for the
+                       endpoint.
+        :return: Dict corresponding to created endpoint, which must contain the 'name' key.
+        """
+        pass
+
+    @abc.abstractmethod
+    def update_endpoint(self, endpoint_name, config=None):
+        """
+        Update the endpoint with the specified name. You can update any target-specific attributes
+        of the endpoint (via `config`). By default, this method should block until the update
+        completes (i.e. until it's possible to create a deployment within the endpoint). See
+        target-specific plugin documentation for additional detail on support for asynchronous
+        update and other configuration.
+
+        :param endpoint_name: Unique name of endpoint to update
+        :param config: (optional) dict containing target-specific configuration for the
+                       endpoint
+        :return: None
+        """
+        pass
+
+    @abc.abstractmethod
+    def delete_endpoint(self, endpoint_name):
+        """
+        Delete the endpoint with name ``name`` from the specified target. Deletion should be
+        idempotent (i.e. deletion should not fail if retried on a non-existent deployment).
+
+        :param endpoint_name: Name of endpoint to delete
+        :return: None
+        """
+        pass
+
+    @abc.abstractmethod
+    def list_endpoints(self):
+        """
+        List endpoints in the specified target. This method is expected to return an
+        unpaginated list of all endpoints (an alternative would be to return a dict with
+        an 'endpoints' field containing the actual endpoints, with plugins able to specify
+        other fields, e.g. a next_page_token field, in the returned dictionary for pagination,
+        and to accept a `pagination_args` argument to this method for passing
+        pagination-related args).
+
+        :return: A list of dicts corresponding to endpoints. Each dict is guaranteed to
+                 contain a 'name' key containing the endpoint name. The other fields of
+                 the returned dictionary and their types may vary across targets.
+        """
+        pass
+
+    @abc.abstractmethod
+    def get_endpoint(self, endpoint_name):
+        """
+        Returns a dictionary describing the specified endpoint, throwing a
+        py:class:`mlflow.exception.MlflowException` if no endpoint exists with the provided
+        name.
+        The dict is guaranteed to contain an 'name' key containing the endpoint name.
+        The other fields of the returned dictionary and their types may vary across targets.
+
+        :param endpoint_name: Name of endpoint to fetch
+        """
+        pass
+
+    @abc.abstractmethod
+    def create_deployment(self, deployment_name, endpoint_name, model_uri, flavor=None, config=None):
```

Comment:
I don't have strong opinion here, but changing
`def create_deployment(self, deployment_name, endpoint_name, ...` to
`def create_deployment(self, name, endpoint, ...` might be cleaner and more future compatible:
Imagine we introduce a URI/URL to identify an endpoint, `endpoint` can still be used in this case.